### PR TITLE
Add cms 1

### PR DESCRIPTION
**Pull request by Aichnerc**
**Checklist:**

 
 - [x] Have you added an explanation of what your changes do and why you'd like them to be included?
 - [ ] Have you updated or added documentation for the change?
 - [x] Have you tested your changes with successful results?

**Type of Changes:**

 
 - [x] Bug fix (non-breaking change which fixes an issue)
 - [ ] New feature (non-breaking change which adds functionality)
 - [x] Breaking change (fix or feature that would cause existing functionality to change)

**What is the current behavior? (link to any open issues here)**
- Currently there is no Firebase integration.
- Currently there is no data being fetched from Firebase.
- Currently the login is not working.
- Currently the HomePage is our template.

**What is the new behavior (if this is a feature change)?**
- Firebase has been initiated.
- The page can now be fetched from Firebase.
- You can now login.
- The HomePage is now the latest published page.

### CODE REVIEW

File: src/store/actions/pageActions.js [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    firestore
+      .collection("pages")
+      .where("uid", "==", uid)
+      .get()
+      .then((querySnapshot) => {
+        let page = querySnapshot.docs.map((doc) => {
+          let data = doc.data();
+
+          data.uid = doc.id;
+          return data;
```
Comment (Aichnerc) [inline src/store/actions/pageActions.js:16]: Add new line.

File: src/store/reducers/pageReducer.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+const pageReducer = (state = initState, action) => {
+  switch (action.type) {
+    case "GET_SUCCESS":
+      return {
+        ...state,
+        fields: action.data,
+      };
+    case "GET_FAIL":
+      console.log("Error", action.err);
```
Comment (Aichnerc) [inline src/store/reducers/pageReducer.js:14]: Is this `console.log();` really needed?

**Review by Aichnerc — COMMENTED** [review]
Mmm. 👀

File: src/store/actions/pageActions.js [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+      })
+      .catch((err) => {
+        dispatch({
+          type: "GET_FAIL",
+          err,
+        });
+      });
+  };
+};
+//#endregion
```
Comment (kleberbaum) [inline src/store/actions/pageActions.js:41]: Please add a newline.

**Review by kleberbaum — CHANGES_REQUESTED** [review]
Should be ok to merge after a few changes.

File: src/store/reducers/pageReducer.js (hunk 2 above)
Comment (Aichnerc) [reply src/store/reducers/pageReducer.js:14]: Yep.

**Review by LucaAllmaier — APPROVED** [review]
Looks good to me.

File: src/components/pages/HomePage/index.jsx [hunk 4]
```diff
@@ -30,118 +43,121 @@ import Projects from "../../../assets/content/projects.jpg";
...
+        <h2>{column.content.head}</h2>
+        <p className="lead">{column.content.subhead}</p>
       </>
     );
+  };
+
+  render() {
+    const { page } = this.props;
+
+    console.log(page);
```
Comment (schettn) [inline src/components/pages/HomePage/index.jsx:63]: Is this still needed?

File: src/components/pages/HomePage/index.jsx [hunk 5]
```diff
@@ -30,118 +43,121 @@ import Projects from "../../../assets/content/projects.jpg";
...
+        <MDBContainer className="flex-center my-5 py-5">
+          <div className="spinner-grow text-danger" role="status">
+            <span className="sr-only">Loading...</span>
+          </div>
+        </MDBContainer>
+      );
+    } else {
+      const sections = page.fields?.sections;
+
+      console.log(sections);
```
Comment (schettn) [inline src/components/pages/HomePage/index.jsx:76]: Is this still needed?

File: src/components/pages/ProfilePage/index.jsx [hunk 6]
```diff
@@ -47,32 +66,101 @@ class ProfilePage extends React.Component {
...
 
+      // Get the latest version
+      const latestVersionTimestamp = Math.max(
+        ...Object.keys(profile.versions).map((version) => {
+          return version;
+        })
+      );
+      const latestVersion = profile.versions[latestVersionTimestamp];
+
+      console.log(latestVersion);
```
Comment (schettn) [inline src/components/pages/ProfilePage/index.jsx:77]: Is this still needed?

File: src/components/pages/ProfilePage/index.jsx [hunk 7]
```diff
@@ -32,8 +37,22 @@ import {
 class ProfilePage extends React.Component {
   state = {};
 
+  componentDidMount = () => {
+    //this.props.getPage();
```
Comment (schettn) [inline src/components/pages/ProfilePage/index.jsx:41]: Is this still needed?

File: src/store/actions/pageActions.js [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+
+          return data;
+        });
+
+        // Get the latest version
+        const latestVersionTimestamp = Math.max(
+          ...Object.keys(page[0].versions).map((version) => {
+            return version;
+          })
+        );
```
Comment (schettn) [inline src/store/actions/pageActions.js:24]: add newline

**Review by schettn — CHANGES_REQUESTED** [review]
Check for console logs and for missing new lines then it should be on a good way.

File: src/components/pages/HomePage/index.jsx (hunk 4 above)
Comment (Aichnerc) [reply src/components/pages/HomePage/index.jsx:63]: Yep

File: src/components/pages/ProfilePage/index.jsx (hunk 6 above)
Comment (Aichnerc) [reply src/components/pages/ProfilePage/index.jsx:77]: Yep

File: src/components/pages/HomePage/index.jsx (hunk 5 above)
Comment (Aichnerc) [reply src/components/pages/HomePage/index.jsx:76]: Yep

File: src/components/pages/ProfilePage/index.jsx (hunk 7 above)
Comment (Aichnerc) [reply src/components/pages/ProfilePage/index.jsx:41]: Yep

**Review by schettn — APPROVED** [review]
Yep

**Review by kleberbaum — APPROVED** [review]
Should be ok to merge.

**Pull request merged**